src/libraries/System.Linq.Expressions/src/System/Linq/Expressions/LambdaExpression.cs

```diff
@@ -102,6 +102,21 @@ internal virtual int ParameterCount
             }
         }
 
+        [DynamicDependency("Compile()", typeof(LambdaExpression))]
+        [DynamicDependency("Compile()", typeof(Expression<>))]
```

Comment:
This looks a bit weird - maybe add a comment that Exception<> redefines `Compile` method and thus having it on base class is not enough.

Reply:
I've added some comments to this method, and also refactored it based on the below comment.